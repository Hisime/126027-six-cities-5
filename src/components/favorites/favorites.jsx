import React from "react";
import {getFavoriteOffers} from "../../store/selectors/selectors";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {offerPropType} from "../../propTypes";
import {Logo} from "../logo/logo";
import UserNav from "../user-nav/user-nav";
import {fetchFavorites} from "../../store/api-actions";
import {AppRoute, LogoType, OfferListType} from "../../consts";
import OfferList from "../offer-list/offer-list";
import {FilterActions} from "../../store/action";
import {FavoritesEmpty} from "../favorites-empty/favorites-empty";

class Favorites extends React.PureComponent {
  constructor(props) {
    super(props);

    this.handleCityClick = this.handleCityClick.bind(this);
  }

  componentDidMount() {
    this.props.getFavoritesData();
  }

  handleCityClick(city) {
    this.props.setCity(city);
    this.props.history.push(AppRoute.MAIN);
  }

  render() {
    const {offersWithCities} = this.props;
    if (!offersWithCities) {
      return (
        <div className="preloader">PRELOADER</div>
      );
    }
    if (Object.keys(offersWithCities).length === 0) {
      return <FavoritesEmpty/>;
    }

    return (
      <div className="page">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <Logo/>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <UserNav/>
                </ul>
              </nav>
            </div>
          </div>
        </header>

        <main className="page__main page__main--favorites">
          <div className="page__favorites-container container">
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <ul className="favorites__list">
                {Object.entries(offersWithCities).map(([cityName, offers], idx) => (
                  <li key={`cityName-${idx}`} className="favorites__locations-items">
                    <div className="favorites__locations locations locations--current">
                      <div className="locations__item">
                        <a className="locations__item-link" onClick={() => {
                          this.handleCityClick(cityName);
                        }}>
                          <span>{cityName}</span>
                        </a>
                      </div>
                    </div>
                    <OfferList offers={offers} type={OfferListType.FAVORITES}/>
                  </li>
                ))}
              </ul>
            </section>
          </div>
        </main>
        <footer className="footer container">
          <Logo type={LogoType.FOOTER}/>
        </footer>
      </div>
    );
  }
}

Favorites.propTypes = {
  getFavoritesData: PropTypes.func.isRequired,
  setCity: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
  offersWithCities: PropTypes.shape({
    offers: PropTypes.arrayOf(offerPropType),
  }),
};


const mapStateToProps = (state) => ({
  offersWithCities: getFavoriteOffers(state),
});

const mapDispatchToProps = (dispatch) => ({
  getFavoritesData() {
    dispatch(fetchFavorites());
  },
  setCity(index) {
    dispatch(FilterActions.setCity(index));
  },
});

export {Favorites};
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
