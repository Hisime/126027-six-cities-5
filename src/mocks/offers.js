import {OfferType} from '../consts';
import {Features} from '../consts';

export default [
  {
    id: 1,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Beautiful & luxurious studio at great location`,
    favorite: true,
    rate: 4.8,
    type: OfferType.APARTMENT,
    bedRoomsCount: 3,
    guestsCount: 4,
    price: 120,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      super: true,
      about: `Friendly and positive girl`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
      Features.CABEL_TV
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Max`,
        rate: 3,
        date: `April 2019`,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Perfect place`
      }
    ],
    coords: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Small & cozy room`,
    favorite: false,
    rate: 4,
    type: OfferType.ROOM,
    bedRoomsCount: 1,
    guestsCount: 2,
    price: 10,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Helena`,
      super: false,
      about: `Just normal host`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Leo`,
        rate: 3,
        date: `April 2019`,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Perfect place`
      }
    ],
    coords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Bright large house, arty, bohemian decor`,
    favorite: false,
    rate: 4.8,
    type: OfferType.HOUSE,
    bedRoomsCount: 2,
    guestsCount: 5,
    price: 1000,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: true,
      about: `I work from home most days, I’m a photographer, so I’m around to help with any queries on the area.`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
      Features.CABEL_TV
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Peter`,
        rate: 4.7,
        date: `April 2019`,
        text: `Mark is a fab host! The house is perfect location for great local spots`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Mark is a fab host! The house is perfect location for great local spots.`
      }
    ],
    coords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: false,
    title: `Bright large house, arty, bohemian decor`,
    favorite: true,
    rate: 3,
    type: OfferType.HOTEL,
    bedRoomsCount: 1,
    guestsCount: 2,
    price: 50,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Hotel room 5min to Tower Bridge`,
      super: true,
      about: `I am very friendly and communicative`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Dan`,
        rate: 4.9,
        date: `April 2018`,
        text: `Great clean place. Very quiet. And a lovely helpful host.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2017`,
        text: `Great clean place. Very quiet. And a lovely helpful host.`
      }
    ],
    coords: [52.3809553943508, 4.939309666406198]
  }
];
