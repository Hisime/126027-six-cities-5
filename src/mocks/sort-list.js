import {SortType} from "../consts";

export default [
  {
    value: SortType.POPULAR,
    label: `Popular`,
  },
  {
    value: SortType.PRICE_ASC,
    label: `Price: low to high`,
  },
  {
    value: SortType.PRICE_DES,
    label: `Price: high to low`,
  },
  {
    value: SortType.TOP_DES,
    label: `Top rated first`,
  },
];
