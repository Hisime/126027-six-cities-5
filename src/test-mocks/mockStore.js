import {createStore} from "redux";
import rootReducer from "../store/reducers/root-reducer";

export const mockStore = createStore(rootReducer);
