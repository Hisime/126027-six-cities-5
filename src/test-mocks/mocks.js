/* eslint-disable */

export const user = {
  avatar_url: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
  email: `adawd@awda.ru`,
  id: 1,
  is_pro: false,
  name: `adawd`,
};

export const match = {
  params: {
    id: `1`,
  },
};

export const citiesMock = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const noop = () => {};
