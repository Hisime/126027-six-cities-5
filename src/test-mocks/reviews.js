export const reviewsMock = [{
  "id": 1,
  "user": {
    "id": 11,
    "is_pro": false,
    "name": `Jack`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
  },
  "rating": 5,
  "comment": `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
  "date": `2020-10-10T13:38:44.753Z`,
}, {
  "id": 2,
  "user": {
    "id": 18,
    "is_pro": true,
    "name": `Sophie`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  },
  "rating": 2,
  "comment": `I stayed here for one night and it was an unpleasant experience.`,
  "date": `2020-10-10T13:38:44.753Z`,
}, {
  "id": 3,
  "user": {
    "id": 19,
    "is_pro": false,
    "name": `Christina`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
  },
  "rating": 4,
  "comment": `I stayed here for one night and it was an unpleasant experience.`,
  "date": `2020-10-10T13:38:44.753Z`,
}];
