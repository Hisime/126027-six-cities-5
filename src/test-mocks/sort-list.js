import {sortTypes} from "../consts";

export const sortListMock = [
  {
    value: sortTypes.POPULAR,
    label: `Popular`,
  },
  {
    value: sortTypes.PRICE_ASC,
    label: `Price: low to high`,
  },
  {
    value: sortTypes.PRICE_DES,
    label: `Price: high to low`,
  },
  {
    value: sortTypes.TOP_DES,
    label: `Top rated first`,
  },
];
