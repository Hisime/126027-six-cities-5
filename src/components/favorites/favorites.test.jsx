import React from "react";
import renderer from "react-test-renderer";
import {Favorites} from "./favorites";
import {noop} from "../../test-mocks/mocks";
import {offersWithCity} from "../../test-mocks/offers-with-city";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history/browser-history";
import {Provider} from "react-redux";
import {mockStore} from "../../test-mocks/mockStore";

it(`Should Favorites render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <Router history={browserHistory}>
            <Favorites
              setCity={noop}
              getFavoritesData={noop}
              offersWithCities={offersWithCity}
            />
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
