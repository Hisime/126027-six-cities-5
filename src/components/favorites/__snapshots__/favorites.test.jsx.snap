// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Favorites render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  onClick={[Function]}
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/66"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        111
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "40%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/66"
                      onClick={[Function]}
                    >
                      The Pondhouse - A Magical Place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    House
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/46"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        293
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/46"
                      onClick={[Function]}
                    >
                      Waterfront with extraordinary view
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Hotel
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/87"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        266
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "60%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/87"
                      onClick={[Function]}
                    >
                      Tile House
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Hotel
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/76"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        119
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/76"
                      onClick={[Function]}
                    >
                      Beautiful & luxurious apartment at great location
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  onClick={[Function]}
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/53"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        931
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/53"
                      onClick={[Function]}
                    >
                      The Joshua Tree House
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    House
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/3"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        218
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/3"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/14"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        232
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "40%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/14"
                      onClick={[Function]}
                    >
                      House in countryside
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private Room
                  </p>
                </div>
              </article>
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/2"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        323
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/2"
                      onClick={[Function]}
                    >
                      Amazing and Extremely Central Flat
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  onClick={[Function]}
                >
                  <span>
                    Dusseldorf
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__image-wrapper favorites__image-wrapper"
                >
                  <a
                    href="/offer/10"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height={150}
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
                      width={150}
                    />
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        125
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/10"
                      onClick={[Function]}
                    >
                      Penthouse, 4-5 rooms + 5 balconies
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
